src/components/Sidebar/Tabs/CustomizationTab/MarkSubTab.tsx

```diff
@@ -0,0 +1,140 @@
+  const [results, setResults] = useState(
+    vegaChartList.map((choice, index) => ({ choice, index }))
+  );
+  const data = useModelState<PlainObject>('graph_data', (data) => ({
+    data,
+  }))[0];
+  const [selectedMark, setSelectedMark] = useState<
+    string | Record<string, any>
+  >(props.spec.mark);
+
+  function handleOnClick(spec: GraphSpec) {
+    const newSpec = produce(spec, (draftSpec: GraphSpec) => {
+      draftSpec['width'] = props.spec.width;
+      draftSpec['height'] = props.spec.height;
+    });
+
+    setSelectedMark(newSpec.mark);
+    props.setSpec(newSpec);
+  }
+
+  return (
+    <div className="mark-options-wrapper" style={{ width: '100%' }}>
+      <SearchBar
+        choices={vegaChartList}
+        onResultsChange={setResults}
+        onChange={setSearchValue}
+        value={searchValue}
+      />
+      <ul className="mark-options" css={markOptionsListCss}>
+        {results.map(({ choice: kind }) => {
+          let shouldSkip = false;
+          const spec = produce(props.spec, (draftSpec: GraphSpec) => {
+            draftSpec['width'] = 100;
+            draftSpec['height'] = 100;
+            if (vegaCategoricalChartList.includes(kind)) {
+              const x = draftSpec.encoding['x'];
+              const y = draftSpec.encoding['y'];
+              if (
+                (x['type'] === 'quantitative' && y['type'] === 'nominal') ||
+                (y['type'] === 'quantitative' && x['type'] === 'nominal')
+              ) {
+                if (kind === 'errorband') {
+                  draftSpec.mark = { type: kind, extent: 'ci', borders: true };
+                } else if (kind === 'errorbar') {
+                  draftSpec.mark = { type: kind, extent: 'ci', ticks: true };
+                } else if (kind === 'arc') {
+                  if (x['type'] === 'quantitative') {
+                    draftSpec['encoding']['theta'] = x;
+                    draftSpec['encoding']['color'] = y;
+                  } else {
+                    draftSpec['encoding']['theta'] = y;
+                    draftSpec['encoding']['color'] = x;
+                  }
+                } else if (kind === 'boxplot') {
+                  draftSpec.mark = { type: kind, extent: 'min-max' };
+                }
+              } else {
+                shouldSkip = true;
+              }
+            } else if (vegaTemporalChartList.includes(kind)) {
+              const xType = draftSpec['encoding']['x']['type'];
+              const yType = draftSpec['encoding']['y']['type'];
+              if (
+                ['temporal', 'ordinal'].includes(xType) &&
+                yType === 'quantitative'
+              ) {
+                draftSpec.mark = kind;
+              } else {
+                shouldSkip = true;
+              }
+            } else {
+              draftSpec.mark = kind;
+            }
+          });
+          if (shouldSkip) {
+            return null;
+          }
```

Comment:
Instead of having an array with null values, we could `filter` before the `map` like this

```
results.filter(kind => vegaCategoricalChartList.includes(kind) || vegaTemporalChartList.includes(kind)).map(...)
```